saleor/graphql/checkout/mutations/checkout_create_from_order.py

```diff
@@ -0,0 +1,425 @@
+        variant_ids_set = set(
+            [line.variant_id for line in order_lines if line.variant_id]
+        )
+        channel_id = (order.channel_id,)
+        channel_id = cast(int, channel_id)
+
+        cls.handle_not_found_variants(
+            order_lines=order_lines, variant_errors=variant_errors
+        )
+
+        order_lines = [line for line in order_lines if line.variant_id]
+
+        variant_ids_set = cls.handle_not_available_variants_in_channel(
+            variant_ids_set=variant_ids_set,
+            channel_id=channel_id,
+            order_lines=order_lines,
+            variant_errors=variant_errors,
+        )
+
+        order_lines = [
+            line for line in order_lines if line.variant_id in variant_ids_set
+        ]
+
+        variant_ids_set = cls.handle_not_published_variants(
+            variant_ids_set=variant_ids_set,
+            channel_id=channel_id,
+            order_lines=order_lines,
+            variant_errors=variant_errors,
+        )
+
+        order_lines = [
+            line for line in order_lines if line.variant_id in variant_ids_set
+        ]
+        variant_ids_set = cls.handle_not_available_variants_for_purchase(
+            variant_ids_set=variant_ids_set,
+            channel_id=channel_id,
+            order_lines=order_lines,
+            variant_errors=variant_errors,
+        )
+
+        order_lines = [
+            line for line in order_lines if line.variant_id in variant_ids_set
+        ]
+        variant_ids_set = cls.handle_variants_exceeding_quantity_limit(
+            variant_ids_set, order_lines, variant_errors, global_quantity_limit
+        )
+
+        order_lines = [
+            line for line in order_lines if line.variant_id in variant_ids_set
+        ]
+        return variant_ids_set, order_lines, variant_errors
+
+    @classmethod
+    def clean_order_lines(
+        cls,
+        order: order_models.Order,
+        order_lines: list[order_models.OrderLine],
+        site: Site,
+    ):
+        (
+            variant_ids_set,
+            available_order_lines,
+            variant_errors,
+        ) = cls.exclude_lines_unavailable_to_purchase(
+            order=order,
+            order_lines=order_lines,
+            global_quantity_limit=site.settings.limit_quantity_per_checkout,
+        )
+
+        variants = list(
+            product_models.ProductVariant.objects.select_related(
+                "product__product_type"
+            ).filter(id__in=variant_ids_set)
+        )
+
+        variants, quantities = get_variants_and_total_quantities(
+            variants,
+            [
+                CheckoutLineData(
+                    variant_id=str(line.variant_id), quantity=line.quantity
+                )
+                for line in order_lines
+            ],
+        )
+        valid_order_lines: list[order_models.OrderLine] = []
+        if order.shipping_address:
+            country = order.shipping_address.country.code
+        else:
```

Comment:
We should also handle the situation when the order is digital and have only `billing_address`. 
Maybe this function could be used here `saleor/order/utils.py:get_order_country`?

Reply:
I think we should use one logic in all places. Due to my knowledge, we should choose a country by priorities:
1. country from shipping address if it exists
2. country from billing address if it exists 
3. country default from the channel

Reply:
@fowczarek  But proposed one is a different flow than currently exists in the checkout flow. If we agree to include the billing address also, then it should be done in a separate PR. IMHO, shipping address + default country sounds good to me. Especially since I don't think that we should use billing addresses in the logic related to stock/warehouses.